pkg/controller/cluster/pools.go

```diff
@@ -103,6 +103,9 @@ func poolSSMatchesSpec(tenant *miniov2.Tenant, pool *miniov2.Pool, ss *appsv1.St
 	if !reflect.DeepEqual(expectedMetadata.Annotations, ss.ObjectMeta.Annotations) {
 		poolMatchesSS = false
 	}
-
+	// Try to detect changes in Env Vars
+	if !miniov2.CompareEnvs(tenant.Spec.Env, ss.Spec.Template.Spec.Containers[0].Env) {
```

Comment:
I think this could return false very time, since not all the environment variables in `volumesPerServer: 1` match  `spec.Env` since we add extra environment variables, such as the webhook to operator and other, perhaps

Reply:
You're right, let me look at another approach.

Reply:
@dvaldivia this is fixed. Can you PTAL again.